src/fixes.rs

```diff
@@ -0,0 +1,147 @@
+use crate::common::*;
+
+mod lowercase_key;
+
+trait Fix {
+    fn name(&self) -> &str;
+
+    fn fix_warnings(
+        &self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        let mut count: usize = 0;
+        for warning in warnings {
+            let line = &mut lines[warning.line_number() - 1];
```

Comment:
Can use the `get_mut` method here?

Reply:
Maybe, I'll take a look.

Reply:
We can, but then we have to use `expect` (the line at this index must exist, if not, we should panic):
```rust
let line = lines.get_mut(warning.line_number() - 1).expect("get line");
```

The first version is shorter.

Reply:
I think `get_mut` is more suitable for cases where it is not known whether an element exists.

Reply:
https://doc.rust-lang.org/std/vec/struct.Vec.html#indexing
> Use get and get_mut if you want to check whether the index is in the Vec.

Reply:
I'd insist to use the `get_mut` method to escape panics:
> However be careful: if you try to access an index which isn't in the Vec, your software will panic!

Reply:
But what can we do except panic in this situation (when there is wrong line number in the warning)?
If we use `expect`, we get panic too.

Reply:
> But what can we do except panic in this situation (when there is wrong line number in the warning)?
> If we use `expect`, we get panic too.

May be return `None` instead use `expect`? What do you think?

Reply:
If the entry is not in the array, it means something unexpected happened, the program does not work correctly, we can’t just ignore it.

Reply:
> If the entry is not in the array, it means something unexpected happened, the program does not work correctly, we can’t just ignore it.
> 
> 

Hmm :thinking:
This is a fix that failed, as I understand it.  We can display some information about this, but not complete the work with panic.

Just leaving the uncorrected lines as they are, although maybe this is a bad solution.

Reply:
The missing index means corrupted data or an error in program logic, not failed fix.
It must always be present, if everything is correct (because warnings contain correct line numbers).

Reply:
Yes, I understand and agree, but I will explain what I mean:
fixes is an additional feature and in the case of a program error, it would seem not bad to save the output of the checks results. 
And additionally printing an error, for example, that the fixes encountered a problem.

It might make sense to think about other options.
Just giving panic to the end user of the cli tool will not be very useful and understandable for him.

Reply:
If you specified the `--fix` flag, you definitely want the files to be fixed, I don't think this is an additional feature in such case.

About panic, I will try to give an analogy (a little exaggerated):
```rust
let mut i = 0;
let v = vec![1, 2, 3, 4, 5];

while i < v.len() {
    // should we panic, if there is no element?
    let item = v[i];

    i += 1;
}
```

Reply:
In fact, as I said, I agree :)
Rust provides this protection for us.

I rather worried about the possibility of a human-readable error output, for users who do not know the internal structure of the `dotenv-linter`.

Reply:
I agree with @DDtKey.
In that case we should print an error instead of panic. Maybe we should use the `Result` enum instead of `Option`.

What do you think?

Reply:
What do you think is the correct behavior in this case?

1. Print error and continue to the next warning.
2. Print error, mark all warnings from this file as unfixed again and continue to the next file.
3. Print error and exit (the current behavior).

Reply:
If we are faced with an incorrect line number, the first option seems to be incorrect (there may be a lot of errors in the output) :thinking: 

**I like option 2**, we skip fixing a specific file, leaving only `Warnings` (because other files can be processed correctly).
But since we have to explicitly report the error, the output should look clear.

The current behavior also seems correct and acceptable if the second introduces a lot of difficulties and we will report in which file specifically the error occurred.

Reply:
@mgrachev, what do you think?

Reply:
Sorry for the late response, I was on holiday 🙂

I like the option 2 more.

Reply:
OK, I will make changes )

Reply:
I have added `get_mut` (option 2).